src/main/host/descriptor/socket/netlink.rs

```diff
@@ -0,0 +1,1185 @@
+use std::io::{Cursor, Read, Write};
+use std::net::Ipv4Addr;
+use std::sync::{Arc, Weak};
+
+use atomic_refcell::AtomicRefCell;
+use neli::{FromBytes, ToBytes};
+use neli::consts::rtnl::{
+    Arphrd, Ifa, IfaF, IfaFFlags, Iff, IffFlags, Ifla,
+    RtAddrFamily, Rtm, RtScope,
+};
+use neli::consts::nl::{NlmF, NlmFFlags, Nlmsg};
+use neli::nl::{Nlmsghdr, NlPayload};
+use neli::rtnl::{Ifaddrmsg, Ifinfomsg, Rtattr};
+use neli::types::{Buffer, RtBuffer};
+use nix::errno::Errno;
+use nix::sys::socket::{MsgFlags, NetlinkAddr, Shutdown};
+use shadow_shim_helper_rs::syscall_types::ForeignPtr;
+
+use crate::core::worker::Worker;
+use crate::cshadow as c;
+use crate::host::descriptor::shared_buf::{
+    BufferHandle, BufferState, ReaderHandle, SharedBuf,
+};
+use crate::host::descriptor::socket::{RecvmsgArgs, RecvmsgReturn, SendmsgArgs, Socket};
+use crate::host::descriptor::{
+    File, FileMode, FileState, FileStatus, OpenFile, StateEventSource,
+    StateListenerFilter, SyscallResult,
+};
+use crate::host::memory_manager::MemoryManager;
+use crate::host::syscall::io::{IoVec, IoVecReader, IoVecWriter};
+use crate::host::syscall_types::SyscallError;
+use crate::network::net_namespace::NetworkNamespace;
+use crate::utility::callback_queue::{CallbackQueue, Handle};
+use crate::utility::sockaddr::SockaddrStorage;
+use crate::utility::HostTreePointer;
+
+// this constant is copied from UNIX_SOCKET_DEFAULT_BUFFER_SIZE
+const NETLINK_SOCKET_DEFAULT_BUFFER_SIZE: u64 = 212_992;
+
+pub struct NetlinkSocket {
+    /// Data and functionality that is general for all states.
+    common: NetlinkSocketCommon,
+    /// State-specific data and functionality.
+    protocol_state: ProtocolState,
+}
+
+impl NetlinkSocket {
+    pub fn new(
+        status: FileStatus,
+        _socket_type: NetlinkSocketType,
+        _family: NetlinkFamily,
+    ) -> Arc<AtomicRefCell<Self>> {
+        Arc::new_cyclic(|weak| {
+            // each socket tracks its own send limit
+            let buffer = SharedBuf::new(usize::MAX);
+            let buffer = Arc::new(AtomicRefCell::new(buffer));
+
+            // Get the IP address of the host
+            let default_ip = Worker::with_active_host(|host| host.default_ip()).unwrap();
+            // All the interface configurations are the same as in the getifaddrs function handler
+            let mut interfaces = Vec::new();
+            interfaces.push(Interface {
+                address: Ipv4Addr::LOCALHOST,
+                label: String::from("lo"),
+                prefix_len: 8,
+                if_type: Arphrd::Loopback,
+                mtu: 65536,
```

Comment:
I don't think shadow uses a different MTU for loopback and eth0 (loopback should be 1500 as well). But maybe also worth opening an issue to increase the MTU on loopback if that's what Linux does.

Reply:
> But maybe also worth opening an issue to increase the MTU on loopback if that's what Linux does.

Yes, that's what Linux does. Try running `ip addr`.

Reply:
Resolved by changing both to `cshadow::CONFIG_MTU`

Reply:
Created #3200.